app/models/miq_ae_instance.rb

```diff
@@ -139,6 +143,12 @@ def self.display_name(number = 1)
     n_('Automate Instance', 'Automate Instances', number)
   end
 
+  def self.find_best_match_by(user, relative_path)
+    domain_ids = user.current_tenant.enabled_domains.collect(&:id)
+    joins(:domain).where("miq_ae_namespaces.id in (?)", domain_ids).order("miq_ae_namespaces.priority DESC")
```

Comment:
I think you can do 

`.where(:miq_ae_namespaces => {:id => domain_ids})`

and avoid the sql fragment with `(?)`